homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,294 @@
+                    vol.Optional(
+                        CONF_OFF_DELAY, default=device_data.get(CONF_OFF_DELAY, 0)
+                    ): int,
+                }
+            )
+
+        if (
+            binary_supported(self._selected_device_object)
+            or cover_supported(self._selected_device_object)
+            or light_supported(self._selected_device_object)
+            or switch_supported(self._selected_device_object)
+        ):
+            data_scheme.update(
+                {
+                    vol.Optional(
+                        CONF_SIGNAL_REPETITIONS,
+                        default=device_data.get(CONF_SIGNAL_REPETITIONS, 1),
+                    ): int,
+                }
+            )
+
+        if (
+            self._selected_device_object.device.packettype
+            == DEVICE_PACKET_TYPE_LIGHTING4
+        ):
+            data_scheme.update(
+                {
+                    vol.Optional(
+                        CONF_DATA_BITS, default=device_data.get(CONF_DATA_BITS, 0)
+                    ): int,
+                    vol.Optional(
+                        CONF_COMMAND_ON,
+                        default=hex(device_data.get(CONF_COMMAND_ON, 0)),
+                    ): str,
+                    vol.Optional(
+                        CONF_COMMAND_OFF,
+                        default=hex(device_data.get(CONF_COMMAND_OFF, 0)),
+                    ): str,
+                }
+            )
+
+        return self.async_show_form(
+            step_id="set_device_options",
+            data_schema=vol.Schema(data_scheme),
+            errors=errors,
+        )
+
+    def _get_device_data(self, entry_id):
+        """Get event code based on device identifier."""
+        event_code = None
+        device_id = None
+        for entry in self._device_entries:
+            if entry.id == entry_id:
+                device_id = next(iter(entry.identifiers))[1:]
+                break
+        for packet_id, entity_info in self._config_entry.data[CONF_DEVICES].items():
+            if tuple(entity_info.get(CONF_DEVICE_ID)) == device_id:
+                event_code = packet_id
+                break
+
+        data = {CONF_EVENT_CODE: event_code, CONF_DEVICE_ID: device_id}
+
+        return data
+
+    @callback
+    def update_config_data(self, global_options=None, devices=None):
+        """Update data in ConfigEntry."""
+        entry_data = self._config_entry.data.copy()
+        if global_options:
+            entry_data.update(global_options)
+        if devices:
+            for event_code, options in devices.items():
+                if options is None:
+                    entry_data[CONF_DEVICES].pop(event_code)
+                else:
+                    entry_data[CONF_DEVICES][event_code] = options
+        self.hass.config_entries.async_update_entry(self._config_entry, data=entry_data)
+        self.hass.async_create_task(async_update_options(self.hass, self._config_entry))
```

Comment:
Register a listener instead.

https://developers.home-assistant.io/docs/config_entries_options_flow_handler#signal-updates

Reply:
I did not do that on purpose. The `async_update_entry` is also called when `automatic_add` is `True` and new devices are discovered. In that case it shouldn't reload the integration.

I did not find a way to distinguish between whether the update comes from the option flow or the event from the device.